### ISSUE TITLE
mu4e-compose-attach-message applied to message with invalid RFC822 characters throws assertion 

### ISSUE THREAD

**Issue by davidcsterratt** [djcb/mu#1019]
## Expected or desired behavior

I'd like to be able to forward messages containing invalid yet common RFC822 characters, such as the em dash as an attachment, using `mu4e-compose-attach-message`. Other people have the same problem:
https://groups.google.com/forum/#!topic/mu-discuss/FJo0GmRl4Uo

## Actual behavior

The email does not send and I get the following backtrace and assertion:

```
Debugger entered--Lisp error: (cl-assertion-failed (save-excursion (goto-char (point-min)) (not (re-search-forward "[^#-\377]" nil t))))
  signal(cl-assertion-failed ((save-excursion (goto-char (point-min)) (not (re-search-forward "[^^@-\377]" nil t)))))
  message-send-mail(nil)
  message-send-via-mail(nil)
  message-send(nil)
  message-send-and-exit(nil)
  call-interactively(message-send-and-exit nil nil)
  command-execute(message-send-and-exit)
```
## Steps to reproduce

1. Find a message encoded in windows-1252 containing an em-dash
2. Capture it from the headers view with "a c"
3. Insert it in a buffer with `mu4e-compose-attach-message`
4. Try sending

## Versions of mu, mu4e/emacs, operating system etc.
Scientific Linux 7
emacs 24.5.2
mu4e 0.9.8

## Any other detail

You can work around this by adding `raw=t` to the `<#part ...  >' tag. Then you are asked if the encoding should be raw when you send. This seems to be how `message-forward-make-body-mime` works. 

I am a bit puzzled as the em-dash in that throws the error appears as `\226` in emacs, yet it is found by the rexexp query `(re-search-forward "[^\000-\377]" nil t)`. However, coding systems are still a bit of a mystery to me...

**Comment by eqyiel** [comment]
The `raw=t` workaround didn't work for me.

But after reading [this](https://groups.google.com/d/msg/mu-discuss/FJo0GmRl4Uo/DNlz-KGzqDcJ) I found one that did:

```elisp
(setq
 mm-content-transfer-encoding-defaults
 ;; these are the default values for mm-content-transfer-encoding-defaults
 '(("text/x-patch" 8bit)
   ("text/.*" qp-or-base64)
   ;; except for this one
   ("message/rfc822" qp-or-base64)  ;; original is ("message/rfc822" 8bit)
   ("application/emacs-lisp" qp-or-base64)
   ("application/x-emacs-lisp" qp-or-base64)
   ("application/x-patch" qp-or-base64)
   (".*" base64)))
```

It's not valid RFC822, but you will be able to send your attachments/messages if they contain troublesome characters.

**Comment by tschwinge** [comment]
I also ran into this problem, in a different context (with Emacs' message mode as used by the notmuch MUA).  I managed to send my message by putting both `encoding=base64` and `raw=t` into the `<#part [...]>` tag.  Unfortunately though, the sent message doesn't display correctly.  ;-/

**Comment by sje30** [comment]
I get the same error as @davidcsterratt 

Adding raw=t allowed the message to be sent, but then I can't open the resulting forwarded message in mu4e!

**Comment by djcb** [comment]
Closed as duplicate of #1012.
